servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/execution/SpringGraphQLContextFactory.kt

```diff
@@ -25,8 +26,13 @@ import org.springframework.web.reactive.function.server.ServerRequest
 abstract class SpringGraphQLContextFactory<out T : SpringGraphQLContext> : GraphQLContextFactory<T, ServerRequest>
 
 /**
- * Basic implementation of [SpringGraphQLContextFactory] that returns [SpringGraphQLContext]
+ * Basic implementation of [SpringGraphQLContextFactory] that populates Apollo tracing header.
  */
-class DefaultSpringGraphQLContextFactory : SpringGraphQLContextFactory<SpringGraphQLContext>() {
-    override suspend fun generateContext(request: ServerRequest) = SpringGraphQLContext(request)
+open class DefaultSpringGraphQLContextFactory : SpringGraphQLContextFactory<SpringGraphQLContext>() {
+    override suspend fun generateContextMap(request: ServerRequest): Map<*, Any> = mutableMapOf<Any, Any>()
+        .also { map ->
+            request.headers().firstHeader(FEDERATED_TRACING_HEADER_NAME)?.let { headerValue ->
```

Comment:
nit: a comment on why we are adding this header will be great.

Reply:
its specified in the KDoc on the class -> `Basic implementation of [SpringGraphQLContextFactory] that populates Apollo tracing header.`